package/MDAnalysis/analysis/base.py

```diff
@@ -40,17 +40,60 @@
 logger = logging.getLogger(__name__)
 
 
+class Results(dict):
+    r"""Container object for storing results.
+
+    Results are extend dictionaries by enabling values to be accessed by key,
+    `results["value_key"]`, or by an attribute, `results.value_key`. 
+    They store all results obatined from an analysis after the ``run`` call.
+
+    The current implementation is similar to the `Bunch` class in `sklearn`.
+
+    Examples
+    --------
+    >>> results = Results(a=1, b=2)
+    >>> results['b']
+    2
+    >>> results.b
+    2
+    >>> results.a = 3
+    >>> results['a']
+    3
+    >>> results.c = [1, 2, 3, 4]
+    >>> results['c']
+    [1, 2, 3, 4]
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(kwargs)
+
+    def __setattr__(self, key, value):
+        self[key] = value
```

Comment:
Should you filter for any `value in dir(dict)` so that people do not define keys that they cannot get with `getattr`? In particular "values" would be an obvious one...